modules/trycompile/src/Runtime.cxx

```diff
@@ -74,17 +115,16 @@ bool SketchRuntime::set_sketch_and_car(SketchObject so, BoardData& bdata, const
 }
 
 bool SketchRuntime::clear() {
-    if (running) {
+    if (status != Status::uninitialized && status != Status::ready) {
         vehicle_dat = nullptr;
         curr_sketch = {};
+        if(thr.joinable())
+            thr.join();
         thr = {};
-        running = false;
+        status = Status::uninitialized;
+        return true;
     }
-    return !running;
-}
-
-void SketchRuntime::launch_thread_unchecked() {
-    thr = std::thread{looper, std::cref(curr_sketch), std::ref(stop_tok), std::ref(running)};
+    return false;
 }
 
 }
```

Comment:
Indent?

Reply:
namespace brackets don't add indent in our clang-format, though clang-formatting this whole pr would be welcome

Reply:
Ran a reformat before squashing